File: Stack/Sort a stack/sortStack.cpp
#include <bits/stdc++.h>
using namespace std;
class SortedStack {
public:
  stack<int> s;
  void sort();
};

void printStack(stack<int> s) {
  while (!s.empty()) {
    printf("%d ", s.top());
    s.pop();
  }
  printf("\n");
}

int main() {
  int t;
  cin >> t;
  while (t--) {
    SortedStack *ss = new SortedStack();
    int n;
    cin >> n;
    for (int i = 0; i < n; i++) {
      int k;
      cin >> k;
      ss->s.push(k);
    }
    ss->sort();
    printStack(ss->s);

    cout << "~" << "\n";
  }
}

void sorted_push(int top, stack<int> &s) {
  if (s.empty() || top > s.top()) {
    s.push(top);
    return;
  }
  int newtop = s.top();
  s.pop();
  sorted_push(top, s);

  s.push(newtop);
}

void SortedStack ::sort() {
  if (s.empty())
    return;
  int top = s.top();
  s.pop();
  sort();
  sorted_push(top, s);
}

File: Stack/Sort the Stack/SortStack.java
import java.util.PriorityQueue;
import java.util.Scanner;
import java.util.Stack;

class SortedStack {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    int t = sc.nextInt();
    while (t-- > 0) {
      Stack<Integer> s = new Stack<>();
      int n = sc.nextInt();
      while (n-- > 0) s.push(sc.nextInt());
      GfG g = new GfG();
      Stack<Integer> a = g.sort(s);
      while (!a.empty()) {
        System.out.print(a.peek() + " ");
        a.pop();
      }
      System.out.println();
      System.out.println("~");
    }
    sc.close();
  }
}

class GfG {
  public Stack<Integer> sort(Stack<Integer> s) {
    Stack<Integer> st = new Stack<>();
    PriorityQueue<Integer> pq = new PriorityQueue<>();
    while (!s.isEmpty()) {
      int i = s.pop();
      pq.add(i);
    }
    while (!pq.isEmpty()) {
      int i = pq.remove();
      st.push(i);
    }
    return st;
  }
}
